src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/CreateCollectionCommand.java

```diff
@@ -70,6 +79,22 @@ public VectorSearchConfig(Integer dimension, String metric) {
       }
     }
 
+    public record IndexingConfig(
+        @JsonInclude(JsonInclude.Include.NON_NULL)
+            @Schema(
+                description = "List of allowed indexing fields",
+                type = SchemaType.ARRAY,
+                implementation = String.class)
+            @Nullable
+            List<String> allow,
+        @JsonInclude(JsonInclude.Include.NON_NULL)
+            @Schema(
+                description = "List of denied indexing fields",
+                type = SchemaType.ARRAY,
+                implementation = String.class)
+            @Nullable
+            List<String> deny) {}
```

Comment:
As per my note below, I think validation could be encapsulated within config class itself; either in (canonical) constructor, or, maybe cleaner, in a `create(...)` method that verifies everything is good (and if not throws exception), then calls canonical constructor.

Reply:
Thought about it., For this need to create separate validators to verify things like either one of the field can only be used. So decided to do it as part of resolver.

Reply:
By validator I didn't mean anything pluggable or related to Quarkus.